Dockerfiles/mlpack-dependencies-debian/Dockerfile

```diff
@@ -11,44 +11,70 @@ RUN apt-get update  && apt-get install -y aptitude && apt-get purge -y \
 
 # Installing dependencies required to run mlpack.
 RUN apt-get update  && apt-get install -y --no-install-recommends wget \
-    libboost-math-dev libboost-program-options-dev libboost-test-dev \
-    libboost-serialization-dev cmake binutils-dev make g++ txt2man git \
-    doxygen liblapack-dev libblas-dev libarpack2 libsuperlu-dev  && \
+    cmake binutils-dev make txt2man git build-essential \
+    doxygen liblapack-dev libblas-dev libarpack2 libsuperlu-dev unzip  && \
     apt-get clean && rm -rf /usr/share/man/?? && rm -rf /usr/share/man/??_* && \
     rm -rf /var/lib/apt/lists/* && rm -rf /usr/share/locale/* && \
     rm -rf /var/cache/debconf/*-old && rm -rf /usr/share/doc/*
 
+#Installing gcc from source
+RUN wget --no-check-certificate\
+    "https://ftp.gnu.org/gnu/gcc/gcc-4.8.1/gcc-4.8.1.tar.gz" \
+    && tar xvzf gcc-4.8.1.tar.gz && rm -r gcc-4.8.1.tar.gz
+WORKDIR gcc-4.8.1
+RUN ./contrib/download_prerequisites && cd .. && mkdir objdir && cd objdir && \
+    $PWD/../gcc-4.8.1/configure --prefix=$PWD --enable-languages=c,c++,fortran \
+    --disable-multilib --disable-bootstrap && make && make install
+ENV PATH /objdir/bin/:$PATH
```

Comment:
You were correct about it. The **make install** installs gcc into /usr/bin only. The custom installation replaces the gcc which came with build-essential. So, when I did apt-get purge gcc, it actually removed our custom version. This thing is taken care of now.

```
RUN wget --no-check-certificate\
    "https://ftp.gnu.org/gnu/gcc/gcc-5.1.0/gcc-5.1.0.tar.gz" \
    && tar xvzf gcc-5.1.0.tar.gz && rm -r gcc-5.1.0.tar.gz
WORKDIR gcc-5.1.0
RUN ./contrib/download_prerequisites && cd .. && mkdir objdir && cd objdir && \
    $PWD/../gcc-5.1.0/configure --prefix=/usr --enable-languages=c,c++,fortran \
    --disable-multilib --disable-bootstrap && make -j8 && make install
```

I deleted the ENV command and the RUN command which removes gcc.